crates/cxx-qt-lib-headers/include/qset.h

```diff
@@ -0,0 +1,100 @@
+// clang-format off
+// SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// clang-format on
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+#pragma once
+
+#include <QtCore/QSet>
+
+#include <QtCore/QDate>
+#include <QtCore/QDateTime>
+// #include <QtCore/QPoint>
+// #include <QtCore/QPointF>
+// #include <QtCore/QRect>
+// #include <QtCore/QRectF>
+// #include <QtCore/QSize>
+// #include <QtCore/QSizeF>
+#include <QtCore/QString>
+#include <QtCore/QTime>
+#include <QtCore/QUrl>
+// #include <QtCore/QVariant>
+// #include <QtGui/QColor>
+
+#include "rust/cxx.h"
+
+template<typename T>
+struct rust::IsRelocatable<QSet<T>> : std::true_type
+{
+};
+
+using QSet_bool = QSet<bool>;
+using QSet_f32 = QSet<float>;
+using QSet_f64 = QSet<double>;
+using QSet_i8 = QSet<qint8>;
+using QSet_i16 = QSet<qint16>;
+using QSet_i32 = QSet<qint32>;
+// using QSet_QColor = QSet<QColor>;
+using QSet_QDate = QSet<QDate>;
+using QSet_QDateTime = QSet<QDateTime>;
+// using QSet_QPoint = QSet<QPoint>;
+// using QSet_QPointF = QSet<QPointF>;
+// using QSet_QRect = QSet<QRect>;
+// using QSet_QRectF = QSet<QRectF>;
+// using QSet_QSize = QSet<QSize>;
+// using QSet_QSizeF = QSet<QSizeF>;
+using QSet_QString = QSet<QString>;
+using QSet_QTime = QSet<QTime>;
+using QSet_QUrl = QSet<QUrl>;
+// using QSet_QVariant = QSet<QVariant>;
+using QSet_u8 = QSet<quint8>;
+using QSet_u16 = QSet<quint16>;
+using QSet_u32 = QSet<quint32>;
+
+namespace rust {
+namespace cxxqtlib1 {
+namespace qset {
+
+// Note on the Rust side we rename the methods to "clear" or "len" etc
+// so we need to put each set in it's own namespace otherwise the generated
+// CXX code collides.
+#define CXX_QT_QSET_METHODS(typeName, name, lowerName)                         \
+  namespace lowerName {                                                             \
```

Comment:
the namespace name can't be `bool` as it conflicts with `bool`, but then are names like `u8` or `boolean` any better? or should we call them something like `cxx_bool` or instead drop the nesting and have `qset_bool` (this seems the best solution?)

Reply:
Changed to `qset_T`